### ISSUE TITLE
GH units not reset upon document change

### ISSUE THREAD

**Issue by teocomi** [specklesystems/speckle-sharp#239]
Flagged by @AlanRynne. It seems the units being passed to the GH converter are not refreshed upon Rhino document change. 
The GH connector should always refresh the active document units before sending/receiving.

**Comment by AlanRynne** [comment]
Neither document change nor unit change. May need to have some discussions about this because checking just before send may result in different units in different objects; as conversion in GH happens in many nodes.

**Comment by teocomi** [comment]
Correct! Let me rephrase: the GH components should always instantiate a new `IConverter` with the active document/units before converting.
